src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -366,14 +422,63 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                     scene.pointerY = this._screenCoordinatesRef.y;
                 }
             }
-            controllerData.pick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
 
-            const pick = controllerData.pick;
+            let pick = null;
+            let hoverAtOrigin = false;
+            let hoverRange = false; 
+            if (controllerData.hoverIndexMeshTip) {
+                let hoverPickInfo = this._pickWithMesh(controllerData.hoverIndexMeshTip, false, this.nearPickPredicate);
+                let hoverGrabInfo = this._pickWithMesh(controllerData.hoverIndexMeshTip, false, this.nearGrabPredicate);
+                if ((hoverPickInfo && hoverPickInfo.hit) || (hoverGrabInfo && hoverGrabInfo.hit)) {
+                    hoverRange = true;
+                } 
+                if (hoverRange) {
+                    controllerData.laserPointer.isVisible = false;
+                }
 
-            if (pick && pick.pickedPoint && pick.hit) {
-                // Update laser state
-                this._updatePointerDistance(controllerData.laserPointer, pick.distance);
+                let nearHover = hoverPickInfo && hoverPickInfo.pickedPoint && hoverPickInfo.hit;
+                if (hoverPickInfo?.pickedPoint) {
+                    hoverAtOrigin = hoverPickInfo.pickedPoint.x === 0 && hoverPickInfo.pickedPoint.y === 0 && hoverPickInfo.pickedPoint.z === 0;
+                }
+                if (nearHover && !hoverAtOrigin) {   
+                    controllerData.nearHover = true;
+                    pick = hoverPickInfo;
+                }
+            }
+
+            if (controllerData.pickIndexMeshTip) {

```

Comment:
This contains roughly the same code as the hover section above. It shouldn't be too hard to pull it out into a lambda, returning the pick info.